Rubberduck.CodeAnalysis/Inspections/Concrete/FunctionReturnValueNotUsedInspection.cs

```diff
@@ -76,12 +76,12 @@ private bool IsRecursive(Declaration function)
         private bool IsReturnValueUsed(Declaration function)
         {
             return (from usage in function.References
-                    where !IsAddressOfCall(usage)
-                    where !IsTypeOfExpression(usage)
-                    where !IsCallStmt(usage)
-                    where !IsLet(usage)
-                    where !IsSet(usage)
-                    select usage).Any(usage => !IsReturnStatement(function, usage));
+                where !IsLet(usage)
+                where !IsSet(usage)
+                where !IsCallStmt(usage)
+                where !IsTypeOfExpression(usage)
+                where !IsAddressOfCall(usage)
```

Comment:
Not sure I like the `where`-chaining here. Why not just `&&` the conditions? Too bad all these methods need to be negated, otherwise method groups could be leveraged. Still, not sure why query syntax is being used here.

Reply:
Just for the record - this is going to be considered out of PR's scope since the whole inspection needs refactoring; see: 

https://github.com/rubberduck-vba/Rubberduck/pull/4784#discussion_r255320600

I will add a `TODO` to flag this, though.